src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -103,6 +103,8 @@ module.exports = function(
 
     const instance = workInProgress.stateNode;
     if (typeof instance.shouldComponentUpdate === 'function') {
+      instance.props = workInProgress.memoizedProps;
+      instance.state = workInProgress.memoizedState;
       const shouldUpdate = instance.shouldComponentUpdate(newProps, newState, newContext);
```

Comment:
Don't we have the same potential problem below for the `ReactPureComponent` state check?
```jsx
!shallowEqual(instance.state, newState)
```

Shouldn't this function always use `memoizedProps` and `memoizedState`? (Can we also just remove the redundant `oldProps` parameter?)